frontend/src/features/env/EnvService.ts

```diff
@@ -23,6 +26,25 @@ const createFeedbackResponsesArray = (
 ) => {
   return feedbackFormFieldsStructure.map((question, i) => {
     const { _id, fieldType } = feedbackForm.form.form_fields[i]
+    // Checkbox answers have to be in an answerArray
+    if (fieldType === BasicField.Checkbox) {
+      // if no checkbox is selected, return an empty array
+      const answerArray: string[] = formInputs[question] || []
+
+      // if Others is selected, remove '!!FORMSG_INTERNAL_CHECKBOX_OTHERS_VALUE!!'
+      // which is the last value in the array when 'Others' is selected,
+      // and replace it with the value entered in the Others Input field
+      if (formInputs[othersInputName]) {
```

Comment:
this will break if we ever have multiple checkboxes with "Others" in this modal right? unlikely but maybe we should just leave a comment here in case we forget

Reply:
yes correct! okie will add a comment

Reply:
[486ada2](https://github.com/opengovsg/FormSG/pull/5582/commits/486ada2f79e65c1e58bba9bbdaf4b8c0a0b262ba)